shell/platform/fuchsia/flutter/tests/product_configuration_unittests.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "flutter/fml/time/time_delta.h"
+#include "flutter/shell/platform/fuchsia/flutter/product_configuration.h"
+
+using namespace flutter_runner;
+
+namespace flutter_runner_test {
+
+class ProductConfigurationTest : public testing::Test {};
+
+TEST_F(ProductConfigurationTest, ValidVsyncOffset) {
```

Comment:
can you add the following tests:
1. Invalid json string.
2. Negative vsync offset.
3. Unexpected vsync offset (larger than 1 second for example deserves a warning).

Reply:
Done